src/components/MemberArea/MemberArea.js

```diff
@@ -16,17 +17,15 @@ function MemberArea({ onOpenModal }) {
   const { isDesktop } = useDeviceType();
   const [isMemberMenuOpen, setIsMemberMenuOpen] = useState(false);
   const { currentUser, isMentor, isAdmin, isAuthenticated, logout } = useUser();
-    const openBecomeMentor = useCallback(
-    () => onOpenModal('Edit Your Profile', <EditProfile />),
-    [onOpenModal]
-  );
-
+  const api = useApi()
   const auth = useAuth()
-
-  console.log({isAuthenticated})
+  const openBecomeMentor = useCallback(
+    () => onOpenModal('Edit Your Profile', <EditProfile api={api} />),
+    [onOpenModal, api]
+  );
 
   const openPendingApplications = () => {
-    onOpenModal('Pending Applications', <PendingApplications />);
+    onOpenModal('Pending Applications', <PendingApplications api={api}/>);
```

Comment:
Can't we get the api using `useApi` inside `PendingApplications` (and other components)?

Reply:
`<PendingApplications>` is a class component; I'm not aware of how to use a hook inside of class components which is why I'm passing it as a prop here.

Reply:
Oh, you are right :) We'll convert it to function components later.
What do you think about take it directly from the [ApiContext](https://github.com/Coding-Coach/find-a-mentor/blob/51e5f30423a6253f4963ed4f54a65cbd63561ce3/src/context/apiContext/ApiContext.tsx#L5)?

Reply:
I haven't ever tried to directly access a context from a class component.  I was under the impression they were incompatible, but I've never tried or even researched it much.

If our goal is to convert these class components to function components, it feels like passing the `api` down 1 level is a reasonable middle ground for now.  Is there a specific problem you're looking to solve by grabbing the context inside of these class components instead of having them passed down by their parent?

Reply:
No a big one. Just hope to stick to the final props interface which we will have after the migration to function component.
It's not that big of a deal anyway. 

[docs](https://reactjs.org/docs/context.html#:~:text=static-,contextType,-%3D%20ThemeContext%3B%0A%20%20render()%20%7B)

Reply:
According to the linked docs, a class component can only consume a single context without resorting to a pattern that injects the context as a prop.  In the case of `<EditProfile />`, it is already using `UserContext` as its `contextType` so the only option seems to be injecting the context as a prop from a parent.  Given this, how do you feel about leaving the implementation I have in place and revisiting all of this when the component is converted to a function component?

(You might already be saying that, but I wasn't sure)

Reply:
Sure. Let's keep the props implementation 👍